File: src/main/java/com/matchFit/post/repository/PostRepositoryCustom.java
package com.matchFit.post.repository;

import com.matchFit.post.entity.Post;
import com.matchFit.post.entity.Sports;
import com.matchFit.user.entity.Gender;
import java.time.LocalDate;
import java.util.Collection;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PostRepositoryCustom {
    Page<Post> findByFilters(Sports sports, Gender gender, LocalDate date, Pageable pageable);
    List<Post> findByFiltersAndIds(Sports sports, Gender gender, LocalDate date, Collection<Long> ids);
}

File: src/main/kotlin/com/matchFit/post/repository/PostRepositoryCustom.kt
package com.matchFit.post.repository

import com.matchFit.post.entity.Post
import com.matchFit.post.entity.Sports
import com.matchFit.user.entity.Gender
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.LocalDate

interface PostRepositoryCustom {
    fun findByFilters(sports: Sports?, gender: Gender?, date: LocalDate?, pageable: Pageable): Page<Post>
    fun findByFiltersAndIds(sports: Sports?, gender: Gender?, date: LocalDate?, ids: Collection<Long>): List<Post>
}
